draft-homburg-deleg-incremental-deleg.md

```diff
@@ -0,0 +1,621 @@
+
+   If the legacy query was send minimised and needs a followup query, then parallel to that followup query a new incremental deleg query will be send, adding a single label to the previous incremental deleg query name.
+   For example if the triggering query is `www.university.ac.example.` and the target zone `example.` and the minimised legacy query name was `ac.example.` (which also resulted in a NOERROR no answer response), then the incremental deleg query to be send along in parallel with the followup legacy query will become `university.ac.example.`
+   Processing of the responses of those two new queries will as described above.
+
+## `_deleg` label presence {#presence}
+
+Absence of the `_deleg` label in a zone is a clear signal that the zone does not contain any incremental deleg delegations.
+Recursive resolvers SHOULD NOT send any additional incremental deleg queries for zones for which it is known that it does not contain the `_deleg` label at the apex.
+The state regarding the presence of the `_deleg` label within a resolver can be "unknown", "known not to be present", or "known to be present".
+
+When the presence of a `_deleg` label is "unknown", a `_deleg` presence test query SHOULD be send in parallel to the next query for the zone.
+The query name for the test query is the `_deleg` label prepended to the apex of zone for which to test presence, with query type A.
+
+The testing query can have three possible outcomes:
+
+1. The `_deleg` label does not exist within the zone, and an NXDOMAIN response is returned.
+
+   The non-existence of the `_deleg` label MUST be cached for the duration indicated by the "minimum" RDATA field of the SOA resource record in the authority section, adjusted to the boundaries for TTL values that the resolver has ({{Section 4 of !RFC8767}}).
+   For the period the non-existence of the `_deleg` label is cached, the label is "known not to be present" and the resolver SHOULD NOT send any (additional) incremental deleg queries.
+
+2. The `_deleg` label does exist within the zone but contains no data.
+   A NOERROR response is returned with no RRs in the answer section.
+
+   The existence of the `_deleg` name MUST be cached for the duration indicated by the "minimum" RDATA field of the SOA resource record in the authority section, adjusted to the resolver's TTL boundaries.
+   For the period the existence of the empty non-terminal at the `_deleg` label is cached, the label is "known to be present" and the resolver MUST send additional incremental deleg queries as described in TODO.
+
+3. The `_deleg` label does exist within the zone and contains data.
+   A NOERROR response is return with an A RRset in the answer section.
+
+   The returned A RRset in the answer section MUST be cached for the duration indicated by the TTL for the RRset, adjusted to the resolver's TTL boundaries.
```

Comment:
I think this should be formulated in a more neutral way. There doesn't seem any reason to cache the A RRset. Instead the resolver should record that the _deleg label is known to be present for a duration indicated by the A RRset's TTL value.